ns1/resource_record.go

```diff
@@ -595,3 +600,44 @@ func metaToMapString(m *data.Meta) map[string]interface{} {
 	}
 	return stringMap
 }
+
+func isJson(s string) bool {
+	var js interface{}
+	return json.Unmarshal([]byte(s), &js) == nil
+}
+
+func subdivisionConverter(s string) (map[string]interface{}, error) {
+	// Get a string in "Country-Subdivision" or Json format, sort and return a map[string]interface{}
+	subdivisionsMap := make(map[string]interface{})
+	subdivisions := strings.Split(s, ",")
+	if isJson(s) {
+		json.Unmarshal([]byte(s), &subdivisionsMap)
+		for k, v := range subdivisionsMap {
+			strSlice := []string{}
+			for _, str := range v.([]interface{}) {
```

Comment:
can you add a check on the type assertion here to avoid a panic?
```
vals, ok := v.([]interface{})
if !ok {
...
}
for _, str := range vals ....
```